File: Sem2/AlgAdat/02_Gyak/node.c
#include <stdio.h>

typedef struct Node {
    int value;
    struct Node *next;
} Node;

void push(Node *stack, int val) {
    Node node;
    node.value = val;
    node.next = NULL;
}

int main(void) {
    Node stack;
    stack.value = 69;
    stack.next = NULL;

    Node child1;
    child1.value = 15;
    child1.next = NULL;
    stack.next = &child1;

    int val = stack.next->value;
    printf("%d\n", val);
    return 0;
}

File: Sem2/AlgAdat/06_gyak/Node.java
public class Node {
    public int key;
    public Node next;

    public Node(int key, Node next) {
        this.key = key;
        this.next = next;
    }

    @Override
    public String toString() {
        StringBuilder sbr = new StringBuilder();
        
        Node p = this;

        while (p != null) {
            sbr.append(p.key + ",");
            p = p.next;
        }
        sbr.append("null");

        return sbr.toString();
    }
}